bin/llvm-kompile-clang

```diff
@@ -218,9 +230,13 @@ if [ "$main" = "static" ]; then
 elif [[ "$main" =~ "python" ]]; then
   # Don't link jemalloc when building a python library; it clashes with the
   # pymalloc implementation that Python expects you to use.
-  all_libraries=("${libraries[@]}" "-lgmp" "-lgmpxx" "-lmpfr" "-lpthread" "-ldl" "-lffi" "-lunwind")
+  all_libraries=("${libraries[@]}" "-lgmp" "-lgmpxx" "-lmpfr" "-lpthread" "-ldl" "-lffi" "$libunwind")
   flags+=("-fPIC" "-shared" "-I${INCDIR}" "-fvisibility=hidden")
 
+  if [[ "$OSTYPE" == "darwin"* ]]; then
+    flags+=("-Wl,-flat_namespace")
+  fi
+
```

Comment:
I'm not sure if I correctly understood what you are doing here:
> The fix for this issue is to drop our usage of `-flat_namespace` for C shared libraries compiled by the backend.

Aren't we doing the opposite here?

Reply:
Previously, the `-flat_namespace` option was enabled everywhere on macOS (C, Python, interpreter binaries). This change pushes the option inwards so it's only enabled for Python libraries where it's legitimately still needed.